closed/src/java.base/share/native/libjncrypto/NativeCrypto.c

```diff
@@ -474,12 +576,299 @@ JNIEXPORT jlong JNICALL Java_jdk_crypto_jniprovider_NativeCrypto_loadCrypto
+        "libcrypto.1.0.0.dylib",            /* 1.0.x library name */
+#elif defined(_WIN32) /* defined(__APPLE__) */
+        "libcrypto-3-x64.dll",              /* 3.x library name */
+        "libcrypto-1_1-x64.dll",            /* 1.1.x library name */
+        "libeay32.dll",                     /* old library name */
+#else /* defined(_WIN32) */
+        "libcrypto.so",                     /* general symlink library name */
+        "libcrypto.so.3",                   /* 3.x library name */
+        "libcrypto.so.1.1",                 /* 1.1.x library name */
+        "libcrypto.so.1.0.0",               /* 1.0.x library name */
+        "libcrypto.so.10",                  /* old library name */
+#endif /* defined(_AIX) */
+    };
+
+    const size_t numOfLibs = sizeof(libNames) / sizeof(libNames[0]);
+#if defined(_AIX)
+    const size_t num_of_generic = 4;
+#elif defined(__linux__) /* defined(_AIX) */
+    const size_t num_of_generic = 1;
+#else /* defined(__linux__) */
+    const size_t num_of_generic = 0;
+#endif /* defined(_AIX) */
+
+    void *result = NULL;
+    void *prevResult = NULL;
+    size_t i = 0;
+    long tempVersion = 0;
+    long previousVersion = 0;
+
+    /* If JAVA_HOME is not null or empty and no library has been loaded yet, try there. */
+    if ((NULL != chomepath) && ('\0' != *chomepath) && (NULL == crypto_library)) {
+        size_t path_len = strlen(chomepath);
+#if defined(_WIN32)
+        static const char pathSuffix[] = "\\bin\\";
+#else /* defined(_WIN32) */
+        static const char pathSuffix[] = "/lib/";
+#endif /* defined(_WIN32) */
+        char *libPath = malloc(path_len + sizeof(pathSuffix));
+        strcpy(libPath, chomepath);
+
+        /* Append the proper directory using a slash or backslash, depending on the operating system. */
+        strcat(libPath, pathSuffix);
+
+        size_t libPath_len = strlen(libPath);
+        if (traceEnabled) {
+            fprintf(stdout, "Attempting to load library bundled with JDK from: %s\n", libPath);
+        }
+
+        for (i = 0; i < numOfLibs; i++) {
+            size_t file_len = strlen(libNames[i]);
+            /* Allocate memory for the new file name with the path. */
+            char *libNameWithPath = (char *)malloc((libPath_len + file_len));
+
+            if (NULL == libNameWithPath) {
+                if (traceEnabled) {
+                    fprintf(stderr, "\tFailed to allocate memory for file name with path.\n");
+                }
+                continue;
+            }
+
+            strcpy(libNameWithPath, libPath);
+            strcat(libNameWithPath, libNames[i]);
+
+            /* Load OpenSSL Crypto library bundled with JDK. */
+            if (traceEnabled) {
+                fprintf(stdout, "\tAttempting to load: %s\n", libNames[i]);
+            }
+            result = load_crypto_library(traceEnabled, libNameWithPath);
+
+            if (NULL != libNameWithPath) {
+                free(libNameWithPath);
+            }
```

Comment:
`libNameWithPath` cannot be `NULL` here (see lines 700-705).

Reply:
Yes, makes sense. Removed check and `free`.

Reply:
Line 717 must stay (it was just checking for `NULL` that was unnecessary).

Reply:
Sorry, my bad. Re-added `free`.